### PR TITLE
Fixed numbering hint

### PR DESCRIPTION
**Pull request by Dherse**
There is a small bug in the numbering hint as noted by `@minigamedev` on Discord:

image

The fix originated here:

image

This fix simply changes the `heading` to the function name and adds a special edge-case for `math.equation` such that it is shown correctly.

### CODE REVIEW

File: crates/typst-library/src/meta/reference.rs [hunk 1]
```diff
@@ -189,8 +190,12 @@ impl Show for RefElem {
                     )
                 })
                 .hint(eco_format!(
-                    "you can enable heading numbering with `#set {}(numbering: \"1.\")`",
-                    elem.func().name()
+                    "you can enable {0} numbering with `#set {0}(numbering: \"1.\")`",
```
Comment (laurmaedje) [inline crates/typst-library/src/meta/reference.rs:193]: I think the first `{}` should unconditionally use the name. "You can set equation numbering" makes more sense than "You can set math.equation numbering"

File: tests/typ/bugs/equation-numbering-reference.typ [hunk 2]
```diff
@@ -0,0 +1,24 @@
+// In this bug, the hint and error messages for an equation
+// being reference mentioned that it was "header" and was
```
Comment (laurmaedje) [inline tests/typ/bugs/equation-numbering-reference.typ:2]:
```suggestion
// being reference mentioned that it was a "heading" and was
```

File: tests/typ/bugs/equation-numbering-reference.typ [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+$
+    Delta = b^2 - 4 a c
+$ <quadratic>
+
+// Error: 14-24 cannot reference equation without numbering
+// Hint: 14-24 you can enable math.equation numbering with `#set math.equation(numbering: "1.")`
+Looks at the @quadratic formula.
+
+---
+#set page(height: 70pt)
```
Comment (laurmaedje) [inline tests/typ/bugs/equation-numbering-reference.typ:17]: Why the second test case? Isn't there a test for equation references somewhere already?

File: crates/typst-library/src/meta/reference.rs (hunk 1 above)
Comment (Dherse) [reply crates/typst-library/src/meta/reference.rs:193]: Yeah I was just changing it!

File: tests/typ/bugs/equation-numbering-reference.typ (hunk 3 above)
Comment (Dherse) [reply tests/typ/bugs/equation-numbering-reference.typ:17]: Fair enough, I just assumed it was good practice to include the "well behaved" test when testing bugs

**Comment by laurmaedje** [comment]
Thanks!

**Pull request merged**